GiTinder.Tests/Models/SettingsTest.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using Xunit;
+
+namespace GiTinder.Tests.Models
+{
+    public class SettingsTest
+    {
+
+        [Fact]
+        public void CanCreateSettingsWithNullUserName()
+        {
+            var settings = SettingsFactory.CreateSettings();
+            Assert.True(settings.UserName == null);
+        }
+
+        [Fact]
+        public void CanSetUserNameWithString()
+        {
+            var settings = SettingsFactory.CreateSettings();
+            settings.UserName = "Pablos";
+            Assert.Equal("Pablos", settings.UserName);
+        }
+
+
+
+
+
+        [Fact]
+        public void CanCreateSettingsWithGivenNameAndDefaultSettings()
+        {
+
+            var testSettings = SettingsFactory.CreateSettingsWithGivenUserNameTest_TomekAndDefaultSettings();
+
+
+            Assert.True(testSettings.UserName == "Test_Tomek");
+            Assert.True(testSettings.EnableNotification == true);
+            Assert.True(testSettings.MaxDistanceInKm == 160);
+
+            Assert.False(testSettings.MaxDistanceInKm == 1);
+            Assert.True(testSettings.PreferredLanguages == "English");
+            Assert.False(testSettings.PreferredLanguages == "Czech");
+            
+        }
+
+
+        //Why this test does not pass?
+
+        [Fact]
+        public void SettingsWithNullUserNameIsNotValid()
+        {
+            var settings = SettingsFactory.CreateSettingsWithNullUserName();
+            Assert.True(ValidateModel(settings).Count == 1);
+        }
+
+
+        //Why this test does not pass?
+        [Fact]
+        public void SettingsWithInvalidUserName()
+        {
+            var settings = SettingsFactory.CreateSettingsWithInValidUserName();
+            Assert.True(ValidateModel(settings).Count == 1);
+        }
```

Comment:
@FilipTeply the reason is exactly that your factories are not working as they supposed to

Reply:
I changed the constructor in Settings as shown below.
BUT when I try to add-migration I get this error:
**No suitable constructor found for entity type 'Settings'. The following parameters could not be bound to properties of the entity: 'PreferredLanguages'.**

Also, I changed the SettingsFactory method and my SettingsTests are all passing.

  public static Settings CreateSettingsWithValidUserName()
        {
            var settings = new Settings("Mock Filip", true, true, 160, new List<Language> { new Language("Java"), new Language("C#") });
            return settings;
        } 

public Settings(string UserName, bool EnableNotification, bool EnableBackgroundSync,
            int MaxDistanceInKm, List<Language> PreferredLanguages)
        {
            this.UserName = UserName;
            this.EnableNotification = EnableNotification;
            this.EnableBackgroundSync = EnableBackgroundSync;
            this.MaxDistanceInKm = MaxDistanceInKm;
            this.PreferredLanguages = PreferredLanguages;
        }